internal/interceptors.go

```diff
@@ -242,3 +248,1210 @@ func (t *WorkflowOutboundCallsInterceptorBase) GetLastCompletionResult(ctx Conte
+		ActivityTaskCanceled(attrs *historypb.ActivityTaskCanceledEventAttributes) error
+		TimerStarted(attrs *historypb.TimerStartedEventAttributes) error
+		TimerFired(attrs *historypb.TimerFiredEventAttributes) error
+		TimerCanceled(attrs *historypb.TimerCanceledEventAttributes) error
+		WorkflowExecutionCancelRequested(attrs *historypb.WorkflowExecutionCancelRequestedEventAttributes) error
+		WorkflowExecutionCanceled(attrs *historypb.WorkflowExecutionCanceledEventAttributes) error
+		MarkerRecorded(attrs *historypb.MarkerRecordedEventAttributes) error
+		WorkflowExecutionSignaled(attrs *historypb.WorkflowExecutionSignaledEventAttributes) error
+		WorkflowExecutionTerminated(attrs *historypb.WorkflowExecutionTerminatedEventAttributes) error
+		RequestCancelExternalWorkflowExecutionInitiated(attrs *historypb.RequestCancelExternalWorkflowExecutionInitiatedEventAttributes) error
+		RequestCancelExternalWorkflowExecutionFailed(attrs *historypb.RequestCancelExternalWorkflowExecutionFailedEventAttributes) error
+		ExternalWorkflowExecutionCancelRequested(attrs *historypb.ExternalWorkflowExecutionCancelRequestedEventAttributes) error
+		SignalExternalWorkflowExecutionInitiated(attrs *historypb.SignalExternalWorkflowExecutionInitiatedEventAttributes) error
+		SignalExternalWorkflowExecutionFailed(attrs *historypb.SignalExternalWorkflowExecutionFailedEventAttributes) error
+		ExternalWorkflowExecutionSignaled(attrs *historypb.ExternalWorkflowExecutionSignaledEventAttributes) error
+		UpsertWorkflowSearchAttributes(attrs *historypb.UpsertWorkflowSearchAttributesEventAttributes) error
+		StartChildWorkflowExecutionInitiated(attrs *historypb.StartChildWorkflowExecutionInitiatedEventAttributes) error
+		StartChildWorkflowExecutionFailed(attrs *historypb.StartChildWorkflowExecutionFailedEventAttributes) error
+		ChildWorkflowExecutionStarted(attrs *historypb.ChildWorkflowExecutionStartedEventAttributes) error
+		ChildWorkflowExecutionCompleted(attrs *historypb.ChildWorkflowExecutionCompletedEventAttributes) error
+		ChildWorkflowExecutionFailed(attrs *historypb.ChildWorkflowExecutionFailedEventAttributes) error
+		ChildWorkflowExecutionCanceled(attrs *historypb.ChildWorkflowExecutionCanceledEventAttributes) error
+		ChildWorkflowExecutionTimedOut(attrs *historypb.ChildWorkflowExecutionTimedOutEventAttributes) error
+		ChildWorkflowExecutionTerminated(attrs *historypb.ChildWorkflowExecutionTerminatedEventAttributes) error
+	}
+
+	ServiceInterceptor struct {
+		RequestResponse RequestResponseInterceptor
+		Command         CommandInterceptor
+		Event           EventInterceptor
+	}
+)
+
+var HeadersCache = map[string]*commonpb.Header{}
+
+type BaseRequestResponseInterceptor struct{}
+
+var _ RequestResponseInterceptor = &BaseRequestResponseInterceptor{}
+
+func (i *BaseRequestResponseInterceptor) SetHeaderForWorkflowExecution(token string, header *commonpb.Header) {
+	HeadersCache[token] = header
```

Comment:
What is the use case for this? This isn't safe for concurrent calls of course yet is updating a global var.